pkg/agent/controller/agent.go

```diff
@@ -547,16 +580,24 @@ func (a *Controller) updateExportedServiceStatus(name, namespace string,
 
 		raw, err := resource.ToUnstructured(toUpdate)
 		if err != nil {
-			return errors.Wrap(err, "error converting resource")
+			err := errors.Wrap(err, "error converting resource")
```

Comment:
the error was possibly logged at the call site. Hence, the added logging of it here, might result in duplicated lines.
Same applies to a similar change ~10 lines below.

Reply:
Its not in case of conflict. RetryOnConflict() retries without logging anything.
So I removed this one and left only the latter.